devito/ir/support/utils.py

```diff
@@ -80,14 +80,16 @@ def build_iterators(mapper):
     for k, v in mapper.items():
         for d, offs in v.items():
             if d.is_Stepping:
-                sub_iterators = iterators.setdefault(d.parent, set())
-                sub_iterators.update({ModuloDimension(d, i, k._time_size)
-                                      for i in offs})
+                values = iterators.setdefault(d.parent, [])
+                for i in sorted(offs):
```

Comment:
So what does this change? Like comapred to #1438

Reply:
I was reordering the `uindices` (sub-iterators) for deterministic code generation too far in the food chain:

https://github.com/devitocodes/devito/pull/1430/files#diff-774365b68457b07c3690671a8c99ae00L33

I can't reorder them there because now you can have

```
xs = 0, xsi = f(xs, ...)
```

and you can't swap `xsi` with `xs` otherwise `xs`, used to define `xsi`, would be undefined

but I still want a unique ordering of stepping iterators, which I'm enforcing here (where it should have been in the first place after all)

Reply:
But does it break #1438? I.e does it prevent to have same order with and without modulo for the standard time case.

Reply:
no, AFAICT in 1438 you're just adding a `key` to `sorted`, so you can rebase that on top of this one once it lands, and just by adding the key back we'll be fine

Reply:
Ok good.